### ISSUE TITLE
How do you access ground truth skeleton data?

### ISSUE THREAD

**Issue by harrydobbs** [marktube/TreePartNet#2]
Hi there, 

Could you please advise how I can access your ground truth skeletons?
Here is what I got so far: 

```
def main():

    f = h5py.File('../data/treepartnet/tree_labeled_train.hdf5','r')


    point_set, normal, labels, pr = f['points'][0], f['normals'][0], f['primitive_id'][0], f['codebook'][0]

    skeleton = point_set - np.expand_dims(labels,-1) * normal 


    skeleton_pcd = o3d.geometry.PointCloud()
    skeleton_pcd.points = o3d.utility.Vector3dVector(skeleton)
    skeleton_pcd.paint_uniform_color((1, 0, 0))

    pcd = o3d.geometry.PointCloud()
    pcd.points = o3d.utility.Vector3dVector(point_set)
    pcd.paint_uniform_color((0, 1, 0))
    
    o3d.visualization.draw_geometries([pcd, skeleton_pcd])
 ```

Kind regards,
Harry

**Comment by marktube** [comment]
Well, the GT skeletons are obtained from the labeled mesh dataset that is not shared yet. We use the axis of each cylinder to get the GT skeleton. The code you show can not access the accurate skeleton.

**Comment by harrydobbs** [comment]
Is there any chance you can please share the labelled dataset?

Cheers,
Harry

**Comment by marktube** [comment]
OK, I'll find time to pack those files. BTW, is there any free net disk?  My Google Drive and OneDrive are full.

**Comment by harrydobbs** [comment]
Ah I am not too sure. I had a quick google and found this: https://ieee-dataport.org/ - it seems to be free if you are an IEEE member. Otherwise perhaps your University has a publicly accessible server for storing data sets?
It would be great to get a hold of your data, so I could create a shared folder on my google drive - that you can upload the data to (however this is more of a temporary solution, as I will likely run out of space in the future).
Thanks again,
Harry

**Comment by harrydobbs** [comment]
Another option would be to store them directly on git-hub (I think it requires all files to be less than 100mb though, so you might need to split the hdf5 file up).

**Comment by marktube** [comment]
Thanks! I just uploaded the skeletons to my OneDrive. You can access by this [link](https://1drv.ms/u/s!Am5H_4z5WHyYdj2bBsp9-9tw_NI?e=AoMWBX). This package contains all GT skeletons for our test data. You can check the filenames in tree_labeled_test.hdf5 by using 'names' key.

> Another option would be to store them directly on git-hub (I think it requires all files to be less than 100mb though, so you might need to split the hdf5 file up).

**Comment by harrydobbs** [comment]
Thank you! Sounds great.

**Comment by harrydobbs** [comment]
Hi @marktube , 

Is it possible that you could please share a data-set that has the associated radius for every skeleton edge available?

Cheers,
Harry

**Comment by marktube** [comment]
Sorry, we didn't generate the associated radius for every skeleton edge. You may estimate the radius by yourself.

> Hi @marktube ,
> 
> Is it possible that you could please share a data-set that has the associated radius for every skeleton edge available?
> 
> Cheers, Harry

**Comment by harrydobbs** [comment]
Hi @marktube 

Do you have GT skeletons for the training data too?

Cheers,
Harry

**Comment by marktube** [comment]
Sorry, we didn't generate for the training data.

> Hi @marktube
> 
> Do you have GT skeletons for the training data too?
> 
> Cheers, Harry

**Comment by harrydobbs** [comment]
Is there any possibility of you sharing the real life tree point clouds (with images?) ? 
Thanks in advance!

**Comment by LB-KAM** [comment]
I have the same issue about the real life tree point clouds with images. I would appreciate it if you can upload the data.
